state/state.go

```diff
@@ -502,6 +502,50 @@ func (s *State) processBatch(ctx context.Context, batchNumber uint64, batchL2Dat
 	return res, err
 }
 
+// ProcessTx send batch with one tx to the executor to get zk counters
+func (s *State) ProcessTx(ctx context.Context, tx types.Transaction, dbTx pgx.Tx) (*ProcessBatchResponse, error) {
```

Comment:
@ToniRamirezM WDYT about this function? I've the feeling that this logic it's already implemented somewhere else